### PR TITLE
ScalaTest 3.2, drop PhantomJS/Java 8/Scala 2.12, add MS Edge, add GithHub actions

### PR DESCRIPTION
**Pull request by cheeseng**
Please do not merge this, trying out sharable workflow as suggested by ihostage.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    uses: playframework/.github/.github/workflows/binary-check.yml@v2
+
+  check-docs:
+    name: Docs
+    uses: playframework/.github/.github/workflows/cmd.yml@v2
+    with:
+      cmd: sbt doc
+
+  test:
+    runs-on: macos-latest
```
Comment (ihostage) [inline .github/workflows/ci.yml:23]: @mkurz WDYT? I think we don't need a Mac OS for check compatibility with Safari. We can use Ubuntu from our reusable workflows and check all other browsers except Safari 😄

Comment (mkurz) [reply .github/workflows/ci.yml:23]: Probably yes, I am looking into this whole thing right now, give me a bit time.

**Comment by mkurz** [comment]
@cheeseng So I took a look and pushed a couple of commits. IMHO this should now be ready. What I did is
* use our workflows that run with Ubuntu intead of macOs. As you can see in the logs it complains about missing drivers for Safarai and Internet Explorer (:roll_eyes:), however the tests don't crash.
* Upgraded to a unstable 2.9 Play milestone release that comes with latest Jackson.
* Dropped Java 8, but test against Java 17
* Dropped Scala 2.12

What do you think? Can you have a look and review my commits? Are you happy with the test output?

@ihostage Please review as well, thanks!

**Comment by cheeseng** [comment]
@mkurz Not an issue for me to skip Safari (just nice to have imho), and let's forget about IE  :), Ubuntu should be able to run Edge though, it is good to go for me!

**Comment by mkurz** [comment]
Thanks a lot @cheeseng.

**Comment by mkurz** [comment]
@ihostage if you want something changed here let me know,we can do that in a a follow up PR.

**Pull request merged**